x-pack/plugin/ql/src/main/java/org/elasticsearch/xpack/ql/querydsl/container/ScriptSort.java

```diff
@@ -13,20 +13,26 @@
 
 public class ScriptSort extends Sort {
 
-    private final ScriptTemplate script;
+    private final ScriptTemplate nullsSortScript;
+    private final ScriptTemplate valueSortScript;
 
     public ScriptSort(ScriptTemplate script, Direction direction, Missing missing) {
         super(direction, missing);
-        this.script = Scripts.nullSafeSort(script);
+        this.nullsSortScript = Scripts.nullsSort(script, missing);
+        this.valueSortScript = Scripts.nullSafeSort(script);
```

Comment:
Instead of using two separate sort clauses, would it be possible to use only one and combine the two sort criterions into one script?